spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/LazyScopePassingSpanSubscriber.java

```diff
@@ -29,48 +29,67 @@
  * @author Marcin Grzejszczak
  * @since 2.0.0
  */
-// TODO: why are we extending AtomicBoolean and not actually using its methods?
-final class LazySpanSubscriber<T> extends AtomicBoolean implements SpanSubscription<T> {
+final class LazyScopePassingSpanSubscriber<T> implements SpanSubscription<T> {
```

Comment:
interface name is `SpanSubscription`  implementation name is `LazyScopePassingSpanSubscriber` looks strange.  
`SpanSubscription`  is both `Subscription` and `CoreSubscriber` probably better name would be `LazyScopePassingSpanOperator`

Reply:
looked at reactor-core    
`Subscription` + `CoreSubscriber` called `InnerOperator` but its implementations most of the time called `*Subscriber` so it is ok   
interface name is confusing

Reply:
roger. will leave it for now. thx